Add details of bucket-vs-database-stored data (#1051)
wandb/docs: content/guides/hosting/data-security/secure-storage-connector.md

* You can specify a sub-path when configuring a bucket, to ensure that W&B does not store any files in a folder at the root of the bucket. It can help you better conform to your organzation's bucket governance policy.
{{% /alert %}}

> ## Central database-stored data vs. bucket-stored data

Reviewer: ```suggestion
## Data stored in the central database vs buckets
```

When using BYOB functionality, certain types of data will be stored in the W&B central database, and other types will be stored in your bucket. 

> ### Database-stored

Reviewer: ```suggestion
### Database
```

- Metadata for users, teams, artifacts, experiments, and projects
- Reports
- Experiment logs
- System metrics

> ## Bucket-stored

Reviewer: ```suggestion
## Buckets
```
